File: lib/app/Models/OrderShipmentDTO.php
<?php
/**
 * OrderShipmentDTO
 */
namespace app\Models;

/**
 * OrderShipmentDTO
 * @description Список посылок.  В параметре может указываться несколько посылок.
 */
class OrderShipmentDTO {

    /** @var int $id Идентификатор посылки, присвоенный Маркетом.*/
    public $id = 0;

    /** @var string $shipmentDate Формат даты: &#x60;ДД-ММ-ГГГГ&#x60;.*/
    public $shipmentDate = "";

    /** @var string $shipmentTime **Только для модели Экспресс**  Время, к которому магазин должен упаковать заказ и перевести его в статус &#x60;READY_TO_SHIP&#x60;. После смены статуса за заказом приедет курьер.  Поле может появиться не сразу. Запрашивайте информацию о заказе в течении 5–10 минут, пока оно не вернется.  Формат времени: 24-часовой, &#x60;ЧЧ:ММ&#x60;.  Если заказ сделан организацией, параметр не возвращается до согласования даты доставки.*/
    public $shipmentTime = "";

    /** @var \app\Models\OrderTrackDTO[]|null $tracks **Только для модели DBS**  Информация для отслеживания перемещений посылки.*/
    public $tracks = null;

    /** @var \app\Models\OrderParcelBoxDTO[]|null $boxes Список грузовых мест.*/
    public $boxes = null;

}

File: lib/Models/OrderShipmentDTO.cpp


#include "OrderShipmentDTO.h"

using namespace Tiny;

OrderShipmentDTO::OrderShipmentDTO()
{
	id = long(0);
	shipmentDate = std::string();
	shipmentTime = std::string();
	tracks = std::list<OrderTrackDTO>();
	boxes = std::list<OrderParcelBoxDTO>();
}

OrderShipmentDTO::OrderShipmentDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

OrderShipmentDTO::~OrderShipmentDTO()
{

}

void
OrderShipmentDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *idKey = "id";

    if(object.has_key(idKey))
    {
        bourne::json value = object[idKey];



        jsonToValue(&id, value, "long");


    }

    const char *shipmentDateKey = "shipmentDate";

    if(object.has_key(shipmentDateKey))
    {
        bourne::json value = object[shipmentDateKey];



        jsonToValue(&shipmentDate, value, "std::string");


    }

    const char *shipmentTimeKey = "shipmentTime";

    if(object.has_key(shipmentTimeKey))
    {
        bourne::json value = object[shipmentTimeKey];



        jsonToValue(&shipmentTime, value, "std::string");


    }

    const char *tracksKey = "tracks";

    if(object.has_key(tracksKey))
    {
        bourne::json value = object[tracksKey];


        std::list<OrderTrackDTO> tracks_list;
        OrderTrackDTO element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            tracks_list.push_back(element);
        }
        tracks = tracks_list;


    }

    const char *boxesKey = "boxes";

    if(object.has_key(boxesKey))
    {
        bourne::json value = object[boxesKey];


        std::list<OrderParcelBoxDTO> boxes_list;
        OrderParcelBoxDTO element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            boxes_list.push_back(element);
        }
        boxes = boxes_list;


    }


}

bourne::json
OrderShipmentDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["id"] = getId();






    object["shipmentDate"] = getShipmentDate();






    object["shipmentTime"] = getShipmentTime();





    std::list<OrderTrackDTO> tracks_list = getTracks();
    bourne::json tracks_arr = bourne::json::array();

    for(auto& var : tracks_list)
    {
        OrderTrackDTO obj = var;
        tracks_arr.append(obj.toJson());
    }
    object["tracks"] = tracks_arr;






    std::list<OrderParcelBoxDTO> boxes_list = getBoxes();
    bourne::json boxes_arr = bourne::json::array();

    for(auto& var : boxes_list)
    {
        OrderParcelBoxDTO obj = var;
        boxes_arr.append(obj.toJson());
    }
    object["boxes"] = boxes_arr;




    return object;

}

long
OrderShipmentDTO::getId()
{
	return id;
}

void
OrderShipmentDTO::setId(long  id)
{
	this->id = id;
}

std::string
OrderShipmentDTO::getShipmentDate()
{
	return shipmentDate;
}

void
OrderShipmentDTO::setShipmentDate(std::string  shipmentDate)
{
	this->shipmentDate = shipmentDate;
}

std::string
OrderShipmentDTO::getShipmentTime()
{
	return shipmentTime;
}

void
OrderShipmentDTO::setShipmentTime(std::string  shipmentTime)
{
	this->shipmentTime = shipmentTime;
}

std::list<OrderTrackDTO>
OrderShipmentDTO::getTracks()
{
	return tracks;
}

void
OrderShipmentDTO::setTracks(std::list <OrderTrackDTO> tracks)
{
	this->tracks = tracks;
}

std::list<OrderParcelBoxDTO>
OrderShipmentDTO::getBoxes()
{
	return boxes;
}

void
OrderShipmentDTO::setBoxes(std::list <OrderParcelBoxDTO> boxes)
{
	this->boxes = boxes;
}



